### ISSUE TITLE
Can't use views?

### ISSUE THREAD

**Issue by dpo** [JuliaSmoothOptimizers/NLPModelsJuMP.jl#7]
```julia
julia> using OptimizationProblems
julia> using NLPModelsJuMP
julia> model = MathProgNLPModel(hs6());
julia> xx = rand(5);
julia> x = @view xx[2:3];
julia> obj(model, x)
ERROR: MethodError: no method matching forward_eval(::Array{Float64,1}, ::Array{Float64,1}, ::Array{ReverseDiffSparse.NodeData,1}, ::SparseMatrixCSC{Bool,Int64}, ::Array{Float64,1}, ::Array{Float64,1}, ::SubArray{Float64,1,Array{Float64,1},Tuple{UnitRange{Int64}},true}, ::Array{Float64,1}, ::Array{Float64,1}, ::Array{Float64,1}; user_operators=ReverseDiffSparse.UserOperatorRegistry(Dict{Symbol,Int64}(), MathProgBase.SolverInterface.AbstractNLPEvaluator[], Dict{Symbol,Int64}(), Any[], Any[], Any[]))
Closest candidates are:
  forward_eval(::Array{T,1}, ::AbstractArray{T,1}, ::AbstractArray{ReverseDiffSparse.NodeData,1}, ::Any, ::Any, ::Any, ::Array{T,1}, ::Any, ::Any, ::Any; user_operators) where T at /Users/dpo/.julia/v0.6/ReverseDiffSparse/src/forward.jl:20
  forward_eval(::Array{T,1}, ::AbstractArray{T,1}, ::AbstractArray{ReverseDiffSparse.NodeData,1}, ::Any, ::Any, ::Any, ::Array{T,1}, ::Any, ::Any) where T at /Users/dpo/.julia/v0.6/ReverseDiffSparse/src/forward.jl:20 got unsupported keyword argument "user_operators"
  forward_eval(::Array{T,1}, ::AbstractArray{T,1}, ::AbstractArray{ReverseDiffSparse.NodeData,1}, ::Any, ::Any, ::Any, ::Array{T,1}, ::Any) where T at /Users/dpo/.julia/v0.6/ReverseDiffSparse/src/forward.jl:20 got unsupported keyword argument "user_operators"
Stacktrace:
 [1] forward_eval_all(::JuMP.NLPEvaluator, ::SubArray{Float64,1,Array{Float64,1},Tuple{UnitRange{Int64}},true}) at /Users/dpo/.julia/v0.6/JuMP/src/nlp.jl:445
 [2] eval_f(::JuMP.NLPEvaluator, ::SubArray{Float64,1,Array{Float64,1},Tuple{UnitRange{Int64}},true}) at /Users/dpo/.julia/v0.6/JuMP/src/nlp.jl:477
 [3] obj(::NLPModelsJuMP.MathProgNLPModel, ::SubArray{Float64,1,Array{Float64,1},Tuple{UnitRange{Int64}},true}) at /Users/dpo/.julia/v0.6/NLPModelsJuMP/src/mpb_model.jl:131
```

**Comment by abelsiqueira** [comment]
It seems not: https://github.com/mlubin/ReverseDiffSparse.jl/blob/master/src/forward.jl#L13 uses `Vector{T}`, and it would need to be `AbstractVetor{T}`. Some fields are Abstract while other not, maybe there is a reason?
These `AbstractVector` were added a short time ago (https://github.com/mlubin/ReverseDiffSparse.jl/pull/53), but not all fields were updated to `Vector{T}`.

**Comment by dpo** [comment]
We should ask @mlubin if there's a reason.

**Comment by mlubin** [comment]
No good reason IIRC. PRs accepted.

**Comment by dpo** [comment]
Thanks @mlubin!

**Comment by dpo** [comment]
Fixed in #8.
